tests/daemon/test-circular-file.c

```diff
@@ -609,11 +611,158 @@ test_circular_file_shrink (Fixture      *fixture,
+   * metadata keyfile, if properly written, is 43 bytes long. This is believed
+   * to be caused by g_file_set_contents() not fsync()ing the temporary file
+   * when the target file does not yet exist; poorly-documented semantics
+   * ensue.
+   *
+   * Regardless of cause, a reasonable recovery path is to re-initialize the
+   * metadata, assuming the file is empty.
+   */
+  g_test_bug ("T19953");
+
+  test_circular_file_broken_metadata_file (fixture, contents, length,
+                                           G_KEY_FILE_ERROR_GROUP_NOT_FOUND);
+}
+
+static void
+test_circular_file_metadata_file_junk (Fixture      *fixture,
+                                       gconstpointer unused)
+{
+  gssize length = 43;
+  gchar contents[length];
+
+  /* Totally broken! */
+  memset (contents, '!', length);
+
+  test_circular_file_broken_metadata_file (fixture, contents, length,
+                                           G_KEY_FILE_ERROR_PARSE);
+}
+
+static void
+test_circular_file_metadata_file_missing_max_size (Fixture      *fixture,
+                                                   gconstpointer unused)
+{
+  const gchar contents[] = "[metadata]\nsize=1024\nhead=0\n";
+  const gssize length = sizeof contents;
+
+  /* max_size is missing; just assume the file is empty, since this will never
+   * happen in practice without other keys also being missing unless someone
+   * edits the file by hand and then you get to keep both pieces.
+   */
+  test_circular_file_broken_metadata_file (fixture, contents, length,
+                                           G_KEY_FILE_ERROR_KEY_NOT_FOUND);
+}
+
+static void
+test_circular_file_metadata_file_missing_size (Fixture      *fixture,
+                                               gconstpointer unused)
+{
+  const gchar contents[] = "[metadata]\nmax_size=1024\nhead=27\n";
+  const gssize length = sizeof contents;
+
+  /* size is missing; we should recover by treating the file as empty because
+   * we do not know at what byte we should loop around.
+   */
+  test_circular_file_broken_metadata_file (fixture, contents, length,
+                                           G_KEY_FILE_ERROR_KEY_NOT_FOUND);
+}
+
+static void
+test_circular_file_metadata_file_missing_head (Fixture      *fixture,
+                                               gconstpointer unused)
+{
+  const gchar contents[] = "[metadata]\nmax_size=1024\nsize=1024\n";
+  const gssize length = sizeof contents;
+
+  /* size is missing; we should recover by treating the file as empty, because
+   * we don't know where the head was supposed to be.
+   */
+  test_circular_file_broken_metadata_file (fixture, contents, length,
+                                           G_KEY_FILE_ERROR_KEY_NOT_FOUND);
+}
+
 gint
 main (gint                argc,
       const gchar * const argv[])
 {
   g_test_init (&argc, (gchar ***) &argv, NULL);
+  g_test_bug_base ("https://phabricator.endlessm.com/");
```

Comment:
nice

Reply:
Yeah, it is nice. If you call `g_test_bug("T19953")` in a test and run the test suite with `--verbose` it shows up in the output:

```
GTest: run: /cache-size-provider/recovers-if-corrupt/nul
(MSG: Bug Reference: https://phabricator.endlessm.com/T19953)
GTest: result: OK
```

Whether it's actually more useful than a comment is another question but it's there so let's use it!